File: src/main/java/me/sul/worldmanager/summonmob/mobtype/Zombie.java
package me.sul.worldmanager.summonmob.mobtype;

import me.sul.customentity.entity.EntityZombie;
import org.bukkit.Location;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Entity;

public class Zombie implements AutoSummonableMob {
    private final double SUMMONING_CHANCE;
    private final int MIN_DISTANCE;
    private final int MAX_DISTANCE;

    public Zombie(FileConfiguration config, String parentNode) {
        parentNode = parentNode + ".zombie";
        SUMMONING_CHANCE = config.getDouble(parentNode + ".summoning-chance");
        MIN_DISTANCE = config.getInt(parentNode + ".min-distance");
        MAX_DISTANCE = config.getInt(parentNode + ".max-distance");
    }

    @Override
    public Entity summonMob(Location loc) {
        Entity entity = new EntityZombie(loc).getBukkitEntity();
        return entity;
    }

    @Override
    public double getSpawnChance() { return SUMMONING_CHANCE; }
    @Override
    public int getMinDistance() { return MIN_DISTANCE; }
    @Override
    public int getMaxDistance() { return MAX_DISTANCE; }
}

File: src/main/kotlin/kr/sul/worldmanager/summonmob/mobtype/Zombie.kt
package kr.sul.worldmanager.summonmob.mobtype

import me.sul.customentity.entity.EntityZombie
import org.bukkit.Location
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.entity.Entity

class Zombie(config: FileConfiguration, parentNodeParam: String) : AutoSummonableMob {
    override val spawnChance: Double
    override val minDistance: Int
    override val maxDistance: Int
    override fun summonMob(loc: Location): Entity {
        return EntityZombie(loc).bukkitEntity
    }

    init {
        var parentNode = parentNodeParam
        parentNode = "$parentNode.zombie"
        spawnChance = config.getDouble("$parentNode.summoning-chance")
        minDistance = config.getInt("$parentNode.min-distance")
        maxDistance = config.getInt("$parentNode.max-distance")
    }
}